tests/sympc/tensor/replicatedshare_tensor_test.py

```diff
@@ -136,3 +140,48 @@ def test_hook_property(get_clients) -> None:
 
     assert (rst.T.shares[0] == x.T).all()
     assert (rst.T.shares[1] == y.T).all()
+
+
+@pytest.mark.parametrize("parties", [3, 5, 11])
+@pytest.mark.parametrize("security", ["malicious", "semi-honest"])
+def test_rst_distribute_reconstruct(get_clients, parties, security) -> None:
+    parties = get_clients(parties)
+    protocol = Falcon(security)
+    session = Session(protocol=protocol, parties=parties)
+    SessionManager.setup_mpc(session)
+
+    secret = 42.32
+
+    a = MPCTensor(secret=secret, session=session)
+
+    assert np.allclose(secret, a.reconstruct(), atol=1e-5)
+
+
+@pytest.mark.parametrize("parties", [2, 5, 11])
+def test_share_distribution(get_clients, parties):
```

Comment:
MAYBE: The test name does not correctly show what is done in the test: ```test_share_distribution_number_shares``` might indicate better what we test?